extensions/Elasticsearch/Elasticsearch/ElasticsearchMemory.cs

```diff
@@ -218,15 +218,20 @@ public async Task<string> UpsertAsync(
         Embedding embedding = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
         var coll = embedding.Data.ToArray();
 
+        //adjust min score for cosine similarity
+        //https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#knn-similarity-search
+        float adjustedMinSimilarityScore = (float)(2 * minRelevance - 1);
+
         var resp = await this._client.SearchAsync<ElasticsearchMemoryRecord>(s =>
-                    s.Index(index)
+                    s.Index(index).Size(limit)
```

Comment:
`qd.k(limit)` already sets a limit, isn't this redundant? should we remove the `limit` param in `qd.k(limit)`?